doc/sql.txt

```diff
@@ -0,0 +1,436 @@
+
+
+    Parameters: ~
+        {name} (string)  the table name.
+
+    Return: ~
+        boolean
+
+
+sql:schema({tbl}, {info})                                       *sql:schema()*
+    get sql table {name} schema, if table doesn't exist then return empty
+    table.
+
+
+    Parameters: ~
+        {tbl}  (string)   the table name
+        {info} (boolean)  whether to return table info. default false.
+
+    Return: ~
+        table: list of keys or keys and their type.
+
+
+sql:create({name}, {schema})                                    *sql:create()*
+    Create a new sqlite db table with {name} based on {schema}. if
+    {schema.ensure} then create only when it doesn't exists. similar to 'create
+    if not exists'
+
+
+    Parameters: ~
+        {name}   (string)  table name
+        {schema} (table)   the table keys/column and their types
+
+    Return: ~
+        boolean
+
+    Usage: ~
+        `db:create("todos", {id = {"int", "primary", "key"}, title = "text"})`
+
+
+sql:drop({name})                                                  *sql:drop()*
+    Create a new sqlite db table with {name} based on {schema}. if
+    {schema.ensure} then create only when it doesn't exists. similar to 'create
+    if not exists'
+
+
+    Parameters: ~
+        {name} (string)  table name
+
+    Return: ~
+        boolean
+
+    Usage: ~
+        `db:drop("todos")`
+
+
+sql:insert()                                                    *sql:insert()*
+    Insert to lua table into sqlite database table.
+
+
+    Return: ~
+        boolean|integer: true incase the table was inserted successfully, and
+        the last inserted row id.
+
+    Usage: ~
+        db:insert("todos", { title = "new todo" })
+
+
+sql:update({tbl}, {specs})                                      *sql:update()*
+    Update table row with where closure and list of values
+
+
+    Parameters: ~
+        {tbl}   (string)  the name of the db table.
+        {specs} (table)   a {spec} or a list of {specs} with where and values
+                          key.
+
+    Return: ~
+        boolean: true incase the table was updated successfully.
+
+    Usage: ~
+        `db:update("todos", { where = { id = "1" }, values = { action = "DONE"
+        }})`
+
+
+sql:delete({tbl}, {specs})                                      *sql:delete()*
+    Delete a {tbl} row/rows based on the {specs} given. if no spec was given,
+    then all the {tbl} content will be deleted.
+
+
+    Parameters: ~
+        {tbl}   (string)  the name of the db table.
+        {specs} (table)   a {spec} or a list of {specs} with where and values
+                          key.
+
+    Return: ~
+        boolean: true if operation is successfully, false otherwise.
+
+    Usage: ~
+        db:delete("todos")
+        db:delete("todos", { where = { id = 1 })
+        db:delete("todos", { where = { id = {1,2,3} })
+
+
+sql:select()                                                    *sql:select()*
```

Comment:
missing args

Reply:
no docs no arguments generated :laughing:

Reply:
a question about this. Shouldn't this work the same as `delete`? but there you also allow a list of specs. Not here